opensafely/check.py

```diff
@@ -0,0 +1,115 @@
+from opensafely._vendor import requests
+from opensafely._vendor.ruamel.yaml import YAML
+import os
+import re
+import glob
+import subprocess
+
+DESCRIPTION = "Check the opensafely project for correctness"
+
+RESTRICTED_DATASETS = {"icnarc": ["admitted_to_icu"]}
+
+# todo: dummy location and data
+PERMISSIONS_URL = "https://raw.githubusercontent.com/Jongmassey/research_repository_permissions/main/repository_permissions.yaml"
```

Comment:
I think we need to find a proper home for this file before we land it.

We could also supply it via environment variable with a default value, which can be useful for testing.

Reply:
https://github.com/opensafely-core feels like the natural organisation for it. It doesn't really belong in this repository, or rather it might be harder for non-developer users to update permissions here.